website/docs/cdktf/create-and-deploy/terraform-cloud.mdx

```diff
@@ -0,0 +1,141 @@
+```
+
+3. Run `cdktf init` and log in to Terraform Cloud.
+
+### VCS-driven Workflow
+
+Check your synthesized code into version control and connect your Terraform Cloud workspace to that repository with the VCS-driven workflow. Refer to [Deployment Patterns](/cdktf/create-and-deploy/deployment-patterns#deploy-with-terraform-and-terraform-cloud) for more details about how to use CDKTF with the VCS-driven workflow.
+
+## Managing Variables and Secrets through Terraform Cloud
+
+Terraform Cloud [variables](/cloud-docs/workspaces/variables) let you define the variables and secrets that Terraform uses during remote operations. You can set variables specifically for each workspace or you can create variable sets to reuse the same variables across multiple workspaces. For example, you could define a variable set of provider credentials and automatically apply it to all of the workspaces using that provider. 
+
+To use variables, set your workspace to [Remote Execution Mode](/cloud-docs/workspaces/settings#execution-mode). Terraform can only access workspace variables when executing remotely on Terraform Cloud. 
+
+Use the `TerraformVariable` construct to declare variables in your CDKTF application. This creates an undefined [Terraform input variable](/language/values/variables). The following example demonstrates how to create a `TerraformVariable` called `my-var`. 
+
+```ts
+import { Construct } from "constructs";
+import { App, TerraformStack, TerraformVariable } from "cdktf";
+
+class MyStack extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+    // You can configure the variable 'my-var' in Terraform Cloud and use the value throughout your CDKTF application
+    const value = new TerraformVariable(this, "my-var", {
+      default: "The default value to use",
+      description: "",
+      nullable: false, // if passing no
+      sensitive: true,
+      type: "string",
+    });
+
+    // define resources here
+  }
+}
+```
+After you declare a variable in your CDKTF application, you must [add it to your Terraform Cloud workspace](/cloud-docs/workspaces/variables/managing-variables#add-a-variable) so that Terraform can access the value during plan, apply, and destroy operations.
+
+### Variable Return Values
+
+The return value for `TerraformVariable` is an object with various token representations of the value as attributes. You can pass this object as a string with `value.stringValue`, as a number with `value.numberValue`, or as a list with `value.listValue`. 
+
+This means that there is no actual value in the variable field when CDKTF synthesizes your application. Instead, CDKTF uses a [Token](/cdktf/concepts/tokens), which represents a value that is unknown until Terraform applies your configuration. You cannot use tokens in dynamic checks during the runtime. For example, `if (value.listValue.length > 42) {` always returns `false` because tokenized lists have a static length of one item.
```

Comment:
```suggestion
This means that there is no actual value in the variable field when CDKTF synthesizes your application. Instead, CDKTF uses a [Token](/cdktf/concepts/tokens), which represents a value that is unknown until Terraform applies your configuration. You cannot use tokens in dynamic checks during runtime. For example, `if (value.listValue.length > 42) {` always returns `false` because tokenized lists have a static length of one item.
```